erts/emulator/beam/jit/x86/instr_common.cpp

```diff
@@ -1187,15 +1187,11 @@ void BeamModuleAssembler::emit_i_is_tuple(const ArgVal &Fail,
         /* Fast path for the `error | {ok, Value}` case. */
         comment("simplified tuple test since the source is always a tuple "
                 "when boxed");
-        a.test(getArgRef(Src, sizeof(byte)),
-               imm(_TAG_PRIMARY_MASK - TAG_PRIMARY_BOXED));
-
-        /* Note: ARG2 will NOT be set. This is OK since the operand
-         * for `current_tuple` has a type; that operand will not match
-         * the type-less operand for `get_tuple_element`. Thus, there
-         * will always be a `load_tuple_ptr` instruction emitted if
-         * this instruction is immediately followed by a
-         * `get_tuple_element` instruction. */
+        /* We must be careful to still leave the pointer to the tuple
+         * in ARG2. */
+        mov_arg(ARG2, Src);
```

Comment:
Maybe hoist this move, and change the comment to explain why we're going through `ARG2`?